src/app/components/harvest/screens/uploading/batch-uploading.component.ts

```diff
@@ -19,15 +43,51 @@ export class BatchUploadingComponent implements OnInit {
     return this.stages.harvest;
   }
 
+  public get project(): Project {
+    return this.stages.project;
+  }
+
   public get loading(): boolean {
     return this.stages.transitioningStage;
   }
 
-  public onCancel(): void {
+  public exampleSite(project: Project): Site {
+    return project.sites[0];
+  }
+
+  public async openCancelModal(): Promise<void> {
+    const ref = this.modal.open(ConfirmationComponent, { centered: true });
+    ref.componentInstance.description =
+      "Are you sure you want to cancel this upload? Any uploaded files will be deleted!";
+    ref.componentInstance.cancelBtn = "Return";
+    ref.componentInstance.nextBtn = "Cancel Upload";
+
+    try {
+      await ref.result;
+      this.stages.transition("complete");
+    } catch (err) {}
+  }
+
+  public async openFinishedUploadingModal(): Promise<void> {
+    const ref = this.modal.open(ConfirmationComponent, { centered: true });
+    ref.componentInstance.description =
+      "Are you sure your upload is finished? You can come back here later, if there are any issues.";
+    ref.componentInstance.cancelBtn = "Cancel";
+    ref.componentInstance.nextBtn = "Scan Files";
+
+    try {
+      await ref.result;
+      this.stages.transition("scanning");
+    } catch (err) {}
+  }
```

Comment:
same comments in the review component. Don't use try/catch, duplicated logic, it'd be better if the string were part of the template.